Source/Core/Core/HW/MMIO.h

```diff
@@ -177,45 +178,30 @@ class Mapping
   HandlerArray<u32>::Write m_write_handlers32;
 
   // Getter functions for the handler arrays.
-  //
-  // TODO:
-  // It would be desirable to clean these methods up using tuples, i.e. doing something like
-  //
-  //     auto handlers = std::tie(m_read_handlers8, m_read_handlers16, m_read_handlers32);
-  //     return std::get<Unit>(handlers)[index];
-  //
-  // However, we cannot use this currently because of a compiler bug in clang, presumably related
-  // to http://llvm.org/bugs/show_bug.cgi?id=18345, due to which the above code makes the compiler
-  // exceed the template recursion depth.
-  // As a workaround, we cast all handlers to the requested one's type. This cast will
-  // compile to a NOP for the returned member variable, but it's necessary to get this
-  // code to compile at all.
   template <typename Unit>
   ReadHandler<Unit>& GetReadHandler(size_t index)
   {
-    static_assert(std::is_same<Unit, u8>::value || std::is_same<Unit, u16>::value ||
-                      std::is_same<Unit, u32>::value,
+    static_assert(std::is_same<Unit, u8>() || std::is_same<Unit, u16>() ||
```

Comment:
@degasus We have for quite a while (`std::make_unique` is C++14, for example). I'd probably keep the static assert, since you get an informative message right away if something is wrong, rather than having to sift through the template error; I can remove it if that's preferable.